helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -151,20 +149,11 @@ private int run0() throws InterruptedException, DockerException {
           .setMaxIntervalMillis(SECONDS.toMillis(30))
           .build().newScheduler();
 
-      long totalMillis = 0;
-
       while (!healthChecker.get().check(containerId)) {
         final long retryMillis = retryScheduler.nextMillis();
         log.warn("container failed healthcheck, will retry in {}ms: {}: {}",
                  retryMillis, config, containerId);
-
-        totalMillis += retryMillis;
-        if (totalMillis >= HEALTHCHECK_MAX_RETRY_MILLIS) {
-          docker.killContainer(containerId);
-          throw new RuntimeException("container failed repeated health checks");
-        } else {
-          Thread.sleep(retryMillis);
-        }
+        Thread.sleep(retryMillis);
```

Comment:
If we health check forever, something at a higher level (i.e. helios-helper) will need to decide when to give up, and what action to take (probably undeploy). I think that's ok, and pretty straight forward to reason about.

But I think things get trickier if we put that logic in the agent.

* If health checks fail, should we let the container keep running? If we do, and it eventually becomes healthy, should we detect that and mark it as RUNNING? 
* Should we stop the container? If so, should helios restart it as if it exited with an error code?
* Should we reuse FAILED (which currently means your container exited with an error code) or introduce HEALTH_CHECK_FAILED?
* If we health check forever, callers can decide when to give up. If the agent gives up should we make the timeout configurable?

I'm inclined to let it health check forever, and revisit if it's proves insufficient, but maybe I'm overthinking this?

@rohansingh @matslina?